### DISCUSSION TITLE
NoScribe on a server

### DISCUSSION THREAD

**Discussion by Tokelu** [kaixxx/noScribe#29] [General]
Hi! 

my organization has been looking for something like NoScribe for some time.
And I've tested it out and must say I'm very impressed with its capabilities.

Our researchers conduct a number of interviews, both with professionals with a number of technical terms, but also with ordinary people and patients, and it seems NoScribe is able to handle it.

There are tools and services for transcribing, but often they are prohibitively expensive and/or make use of cloud-based solutions.
Often these interviews contain personally sensitive information, which is why it is critical that data does not end up on a cloud, or in the hands of companies that have no right to it.

So I have a question that I hope you can/will help with.

I find NoScribe easy to use. But some of our users are not ... technically inclined ... and in addition, many of them are on computers where there are restrictions that would not allow them to install programs.
So I had an idea to put NoScribe on a server, create a web interface where users can upload an audio source, and when the process is finished it sends the transcription to the user.

Under "Development and Contribution" you write something about "If you want to run noScribe directly from the source..." So I assume that you don't necessarily need to install and run the .exe application you refer to on SWITCHdrive. And that you can instead run your .py files directly?

But I can't quite figure out which .py I would have to start and where I should specify settings and source/result files etc.
So you were hoping that you might be able to point me in the right direction?

Thanx

**Comment by kaixxx** [comment]
Hi

Nice to hear that you like noScribe! 
All the source files are here in this repository. "[noScribe.py](https://github.com/kaixxx/noScribe/blob/main/noScribe.py)" is the main script. Under "enironment" you'll find information about the other libraries you need to install. 
That beeing said, your idea is not as simple as you might think:
- noScribe uses advanced ai-tools. It is not easy to set up the environment and get everything running right. 
- noScribe is build as a gui-app. Transforming it into a server script is certainly possible but would require a major rework. You'll need good python knowledge, time and patience for such an endevor. It is not as simple as running noScribe.py on a server - this would simply open the gui on that machine.

In my university I've talked with the it-department and they agreed to distribute the app within the organization. You may try the same in your organization. If you have any questions, feel free to ask.
